modules/data.land/R/soilgrids_soc_extraction.R

```diff
@@ -0,0 +1,149 @@
+  ocd_fit$siteid <- id_fit$value
+  dat <- split(ocd_fit, list(ocd_fit$siteid, ocd_fit$Depth))
+  
+  #assume the ocd profile follows gamma distribution best
+  cgamma <- function(theta, val, stat) {
+    pred <- rep(NA, 4)
+    names(pred) = stat
+    if ("Mean" %in% stat) {
+      pred["Mean"] <- theta[1] / theta[2]
+    }
+    qstat <- as.numeric(stat)[!is.na(as.numeric(stat))]
+    pred[as.character(qstat)] <- qgamma(qstat, theta[1], theta[2])
+    return(sum((pred - val) ^ 2))
+  }
+  
+  fitQ <- function(x) {
+    val = x$Value
+    stat = as.character(x$Quantile)
+    theta = c(10, 10)
+    fit <-
+      list(Gamma = optim(theta, cgamma, val = val, stat = stat))
+    SS <- sapply(fit, function(f) {
+      f$value
+    })
+    par <- sapply(fit, function(f) {
+      f$par
+    })
+    return(list(par = par, SS = SS))
+  }
+  
+  score <- suppressWarnings(lapply(dat, fitQ))
+  bestPar <- sapply(score, function(f) { f$par })
+  mean <- bestPar[1,] / bestPar[2,]
+  std <- sqrt(bestPar[1,] / bestPar[2,] ^ 2)
+  mean_site <- matrix(mean, length(lon_input), 6)
+  rownames(mean_site) <- paste0("Site_", 1:length(lon_input))
+  colnames(mean_site) <- c("0-5cm",
+                           "5-15cm",
+                           "15-30cm",
+                           "30-60cm",
+                           "60-100cm",
+                           "100-200cm")
+  std_site <- matrix(std, length(lon_input), 6)
+  rownames(std_site) <- paste0("Site_", 1:length(lon_input))
+  colnames(std_site) <- c("0-5cm",
+                          "5-15cm",
+                          "15-30cm",
+                          "30-60cm",
+                          "60-100cm",
+                          "100-200cm")
+  #calculate organic carbon stock (ocs) as the sum of organic carbon density multiplied by layer thickness, the unit of ocs is kg/m2, based on Eq. (6）in paper https://www.sciencedirect.com/science/article/pii/S2215016122000462
+  ocs_sum <- mean_site[,1]*(5-0)*0.01+mean_site[,2]*(15-5)*0.01+mean_site[,3]*(30-15)*0.01+mean_site[,4]*(60-30)*0.01+mean_site[,5]*(100-60)*0.01+mean_site[,6]*(200-100)*0.01 
+  #calculate standard deviation of ocs as the square root of sum of variance of layer-specific ocs, the unit of ocs is kg/m2, based on Eq. (8) in paper https://www.sciencedirect.com/science/article/pii/S2215016122000462, except the correlation term due to the lack of information 
+  ocs_std <- sqrt((std_site[,1]*(5-0)*0.01)^2+(std_site[,2]*(15-5)*0.01)^2+(std_site[,3]*(30-15)*0.01)^2+(std_site[,4]*(60-30)*0.01)^2+(std_site[,5]*(100-60)*0.01)^2+(std_site[,6]*(200-100)*0.01)^2)
+  
+  if (!is.null(outdir)) {
+    PEcAn.logger::logger.info(paste0("Storing results in: ",file.path(outdir,"soilgrids_soc_data.RData")))
+    if (! file.exists(outdir)) dir.create(outdir,recursive=TRUE)
+    soilgrids_soc_data <- list("Total_soc" = ocs_sum, "Sdev_soc" = ocs_std)
+    save(soilgrids_soc_data, file = file.path(outdir,"soilgrids_soc_data.RData"))
```

Comment:
Given the size and simplicity of this object, why not just save it as a (more portable) CSV?

Reply:
OK, thanks fine to do.

Again it needs to be: Site_ID, Site_Name, Latitude, Longitude, Total_OCS SD_OCS 

Or something like that for both the memory output and the CSV.  Its better if the result is a cleaner format that we can then transform into what we need